stl/inc/sstream

```diff
@@ -100,6 +163,55 @@ public:
         _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
     }
 
+#if _HAS_CXX20
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    _NODISCARD basic_string<_Elem, _Traits, _Alloc2> str(const _Alloc2& _Al) const {
+        return basic_string<_Elem, _Traits, _Alloc2>{view(), _Al};
+    }
+
+    // The buffer cannot be moved to a string directly, because
+    // the buffer may already be full, and the terminating char is not '\0'.
+    // In that case, copy the string as usual.
+    _NODISCARD _Mystr str() && {
+        _Mystr _Result{_String_constructor_rvalue_allocator_tag{}, _STD move(_Al)};
+        const auto _View = _Get_buffer_view();
+        if (_View._Size >= _View._Res) {
+            // the buffer is full
+            _Result.assign(_View._Ptr, _View._Size);
+        } else {
+            _Result._Move_assign_from_buffer(_View._Ptr, _View._Size, _View._Res);
+            _Mystate &= ~_Allocated;
+        }
+        _Tidy();
+        return _Result;
+    }
+
+    _NODISCARD _Mystr_view view() const noexcept {
+        const auto _View = _Get_buffer_view();
+        if (_View._Ptr) {
+            return _Mystr_view{_View._Ptr, _View._Size};

```

Comment:
If _Ptr can be nullptr but _Size be nonzero that suggests a bug somewhere?

Reply:
Yes. If anywhere is OK, _Size cannot be nonzero when _Ptr is nullptr.

Reply:
Then that suggests the test this is doing is never going to occur?

That is if this can't be just `return _Mystr_view{_View._Ptr, _View._Size};` rather than needing an `if (_View._Ptr)` test that suggests a bug in `_Get_buffer_view`.

Reply:
It may occur. When the stringbuf is initialized with const lvalue empty string, the buffer isn't allocated, and _Ptr will be nullptr.

Reply:
Then _Size will also be 0, so the if test doesn't do anything. Returning `{}` is the same as returning `{nullptr, 0}`.